File: scripts/verify_mcp.js
#!/usr/bin/env node
// MCP liveness probe for obsidian-mcp bridge
// On Windows, npx is a .cmd file, not a binary — use cmd.exe /c to launch it
const { spawn } = require('child_process');

const CMD_EXE = 'cmd.exe';
const CMD_ARGS = ['/c', 'npx', '-y', 'obsidian-mcp', 'D:/document/Dina'];

const VAULT_NAME = 'dina';

function send(proc, msg) {
  proc.stdin.write(JSON.stringify(msg) + '\n');
}

function readLine(proc, timeoutMs) {
  return new Promise((resolve) => {
    const timer = setTimeout(() => resolve(null), timeoutMs);
    proc.stdout.on('data', (data) => {
      clearTimeout(timer);
      proc.stdout.removeAllListeners('data');
      const lines = data.toString().trim().split('\n');
      for (const line of lines) {
        try {
          resolve(JSON.parse(line.trim()));
          return;
        } catch (e) {}
      }
    });
  });
}

async function main() {
  console.log('MCP Bridge Liveness Check');
  console.log('='.repeat(44));

  const proc = spawn(CMD_EXE, CMD_ARGS, { stdio: ['pipe', 'pipe', 'pipe'] });

  send(proc, { jsonrpc: '2.0', id: 1, method: 'initialize', params: { protocolVersion: '2024-11-05', capabilities: {}, clientInfo: { name: 'verify-mcp', version: '1.0' } } });

  const initResult = await readLine(proc, 5000);
  const initOk = initResult && !!initResult.result;
  console.log('Initialized:     ', initOk ? 'OK' : 'FAIL');

  send(proc, { jsonrpc: '2.0', method: 'notifications/initialized', params: {} });

  send(proc, { jsonrpc: '2.0', id: 2, method: 'tools/list', params: {} });
  const toolsResult = await readLine(proc, 5000);
  const tools = (toolsResult && toolsResult.result && toolsResult.result.tools) || [];
  console.log('Tools registered:', tools.length);

  send(proc, { jsonrpc: '2.0', id: 3, method: 'tools/call', params: { name: 'list-available-vaults', arguments: {} } });
  const vaultResult = await readLine(proc, 5000);
  let vault = 'unknown';
  if (vaultResult && vaultResult.result && vaultResult.result.content) {
    for (const c of vaultResult.result.content) {
      if (c.type === 'text') {
        const lines = c.text.split('\n').filter(l => l.trim());
        if (lines.length) vault = lines[lines.length - 1].trim().toLowerCase();
      }
    }
  }
  console.log('Vault:           ', vault);

  proc.stdin.end();
  setTimeout(() => proc.kill(), 500);

  const healthy = initOk && tools.length >= 10 && vault === VAULT_NAME;
  console.log('='.repeat(44));
  console.log('Overall:         ', healthy ? 'HEALTHY' : 'ISSUES');
  process.exit(healthy ? 0 : 1);
}

main().catch(e => { console.error(e); process.exit(1); });

File: scripts/verify_mcp.py
#!/usr/bin/env python3
"""MCP liveness probe for obsidian-mcp bridge.

Lightweight check: verifies npx is available, obsidian-mcp package is installed,
and the vault directory exists and is a valid Obsidian vault — without launching
the long-running MCP server process.
"""
import subprocess, sys, os

VAULT = "D:/document/Dina"

def run(cmd, timeout=10):
    if sys.platform == "win32":
        cmd = ["cmd.exe", "/c"] + cmd
    try:
        p = subprocess.run(cmd, capture_output=True, text=True, timeout=timeout)
        return p.stdout + p.stderr, p.returncode
    except (subprocess.TimeoutExpired, FileNotFoundError):
        return "", 1

# 1. Check npx is available
out, rc = run(["npx", "--version"])
if rc != 0 or not out.strip():
    print("FAIL: npx not found")
    sys.exit(1)
print(f"OK: npx v{out.strip()}")

# 2. Check obsidian-mcp package is installed (quick, exits immediately)
out, rc = run(["npx", "obsidian-mcp"])
if rc != 0 and "No vault paths provided" not in out:
    print(f"FAIL: obsidian-mcp check failed: {out[:200]}")
    sys.exit(1)
print("OK: obsidian-mcp package available")

# 3. Check vault directory exists and is a valid Obsidian vault (has .obsidian dir)
vault_path = "D:\\document\\Dina"
if not os.path.isdir(vault_path):
    print(f"FAIL: vault directory does not exist: {vault_path}")
    sys.exit(1)

obsidian_dir = os.path.join(vault_path, ".obsidian")
if not os.path.isdir(obsidian_dir):
    print(f"WARN: .obsidian directory not found in {vault_path} — vault may not be initialized")
else:
    print(f"OK: vault directory exists and is initialized")

# 4. Quick connectivity check — try running obsidian-mcp briefly and kill it
# Use subprocess.Popen so we can kill it after catching initial output
import signal
try:
    if sys.platform == "win32":
        cmd = ["cmd.exe", "/c", "npx", "-y", "obsidian-mcp", VAULT]
    else:
        cmd = ["npx", "-y", "obsidian-mcp", VAULT]
    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
    # Wait briefly for initial output (tool registration happens fast)
    import time
    time.sleep(1)
    # Check if process is still running (it will be — it's a server)
    poll = p.poll()
    if poll is None:
        # Server started and is running — this is healthy
        p.kill()
        p.wait(timeout=3)
        print("OK: obsidian-mcp launched and registered tools successfully")
    else:
        # Process exited — check if it failed
        stdout, stderr = p.communicate(timeout=3)
        if "Vault directory does not exist" in (stdout + stderr):
            print("FAIL: vault path rejected by obsidian-mcp")
            sys.exit(1)
        print("OK: obsidian-mcp exited cleanly")
except Exception as e:
    print(f"WARN: connectivity check inconclusive: {e}")

print(f"Vault path: {VAULT}")
print("MCP Bridge: HEALTHY")
sys.exit(0)